app/controllers/application_controller.rb

```diff
@@ -206,42 +206,29 @@ def needs_redirect?(host, domain)
   # 3. Otherwise nil
   #
   def self.default_community_fetch_strategy(domain)
-    # Find by domain
-    by_domain = Community.find_by_domain(domain)
+    raise ArgumentError("Domain cannot be nil.") if domain.nil?
 
-    if by_domain.present?
-      return by_domain
-    end
-
-    # Find by username
     app_domain = URLUtils.strip_port_from_host(APP_CONFIG.domain)
     ident = domain.chomp(".#{app_domain}")
-    by_ident = Community.where(ident: ident).first
 
-    if by_ident.present?
-      return by_ident
-    end
+    community = Community.where("domain = ? OR ident = ?", domain, ident).first
```

Comment:
The previous version favours domain but I guess it's not specified here if ident or domain match comes first in the result set. If I get this right you'd need to match with "domain = ? OR (domain != ? AND ident = ?)" to get the same effect.